File: v3/csharp/src/Newscatcherapi.Net/Api/SearchSimilarApi.cs
using System;
using System.Reflection;
using System.Collections.Generic;
using Newscatcherapi.Net.Client;
using Newscatcherapi.Net.Model;
using Newscatcherapi.Net.Client;

namespace Newscatcherapi.Net.Api
{
    public partial class SearchSimilarApi : SearchSimilarApiGenerated {
        public SearchSimilarApi() : base() {}
        public SearchSimilarApi(string basePath): base(basePath) {}
        public SearchSimilarApi(Newscatcherapi.Net.Client.Configuration configuration): base(configuration) {}
        public SearchSimilarApi(Newscatcherapi.Net.Client.ISynchronousClient client, Newscatcherapi.Net.Client.IAsynchronousClient asyncClient, Newscatcherapi.Net.Client.IReadableConfiguration configuration): base(client, asyncClient, configuration) {}
    
        public List<Object> GetAll(Dictionary<string, object> parameters) {
            int pageSize = parameters.ContainsKey("pageSize") ? (int) parameters["pageSize"] : 100;

            Type type = this.GetType();
            MethodInfo method = type.GetMethod("Get");
            List<object> paramValues = SearchAllHelper.prepareParameters(parameters, method);

            try {
                SearchSimilarGetResponse rawResponse = (SearchSimilarGetResponse) method.Invoke(this, paramValues.ToArray());
                DtoResponsesMoreLikeThisResponseSearchResponse response = rawResponse.GetDtoResponsesMoreLikeThisResponseSearchResponse();
                if (response.TotalHits > pageSize)
                {
                    for (int page = 2; page <= response.TotalPages; page++)
                    {
                        parameters["page"] = page;
                        SearchSimilarGetResponse rawRes = (SearchSimilarGetResponse) method.Invoke(this, paramValues.ToArray());
                        DtoResponsesMoreLikeThisResponseSearchResponse res = rawRes.GetDtoResponsesMoreLikeThisResponseSearchResponse();
                        response.Articles.AddRange(res.Articles);
                    }
                }
                return response.Articles;
            } catch (Exception e) {
                Console.WriteLine($"Exception when calling Search API: {e}");
                return null;
            }
        }

        public List<Object> PostAll(MoreLikeThisRequest request) {
            int pageSize = request.PageSize;
            try {
                SearchSimilarPostResponse rawResponse = this.Post(request);
                DtoResponsesMoreLikeThisResponseSearchResponse response = rawResponse.GetDtoResponsesMoreLikeThisResponseSearchResponse();
                if (response.TotalHits > pageSize)
                {
                    for (int page = 2; page <= response.TotalPages; page++)
                    {
                        request.Page = page;
                        SearchSimilarPostResponse rawRes = this.Post(request);
                        DtoResponsesMoreLikeThisResponseSearchResponse res = rawRes.GetDtoResponsesMoreLikeThisResponseSearchResponse();
                        response.Articles.AddRange(res.Articles);
                    }
                }
                return response.Articles;
            } catch (Exception e) {
                Console.WriteLine($"Exception when calling Search API: {e}");
                return null;
            }
        }
    }
}

File: v3/java/src/main/java/com/konfigthis/newscatcherapi/client/api/SearchSimilarApi.java
package com.konfigthis.newscatcherapi.client.api;

import com.konfigthis.newscatcherapi.client.ApiClient;
import com.konfigthis.newscatcherapi.client.ApiException;
import com.konfigthis.newscatcherapi.client.model.*;

import java.util.List;

public class SearchSimilarApi extends SearchSimilarApiGenerated {

    public SearchSimilarApi(ApiClient apiClient) throws IllegalArgumentException {
        super(apiClient);
    }
    public class GetRequestBuilder extends GetRequestBuilderGenerated {
        public GetRequestBuilder(String q) {
            super(q);
        }

        public List<ArticlesPropertyInner1> executeSearchAll() {
            int pageSize = this.pageSize != null ? this.pageSize : 100;
            try {
                FSearchResponse2 response = this.execute();
                if (response.getTotalHits() > pageSize) {
                    for (int page = 2; page <= response.getTotalPages(); page++) {
                        this.page(page);
                        FSearchResponse2 res = this.execute();
                        response.getArticles().addAll(res.getArticles());
                    }
                }
                return response.getArticles();
            } catch (ApiException e) {
                System.out.printf("Exception when calling Search all: %s\n", e);
                return null;
            }
        }
    }
    public class PostRequestBuilder extends PostRequestBuilderGenerated {
        public PostRequestBuilder(String q) {
            super(q);
        }

        public List<ArticlesPropertyInner1> executeSearchAll() {
            int pageSize = this.pageSize != null ? this.pageSize : 100;
            try {
                FSearchResponse3 response = this.execute();
                if (response.getTotalHits() > pageSize) {
                    for (int page = 2; page <= response.getTotalPages(); page++) {
                        this.page(page);
                        FSearchResponse3 res = this.execute();
                        response.getArticles().addAll(res.getArticles());
                    }
                }
                return response.getArticles();
            } catch (ApiException e) {
                System.out.printf("Exception when calling Search all: %s\n", e);
                return null;
            }
        }
    }
}
